src/refract/PrintVisitor.cc

```diff
@@ -10,14 +10,18 @@
 #include <sstream>
 #include <iostream>
 
-#define VISIT_IMPL( ELEMENT ) void PrintVisitor::visit(const ELEMENT ## Element& e) \
+#include "PrintVisitor.h"
+#include "Visitor.h"
+
+#define VISIT_IMPL( ELEMENT ) void PrintVisitor::operator()(const ELEMENT ## Element& e) \
     {                                                                    \
         typedef ELEMENT ## Element::ValueType::const_iterator iterator;  \
         PrintVisitor ps(indent + 1, os);                                 \
+        Visitor visitor(ps);                                             \
                                                                          \
         os << "" #ELEMENT "Element  {\n";                                \
         for (iterator it = e.value.begin(); it != e.value.end(); ++it) { \
-            ps.visit(*(*it));                                            \
+            visitor.visit(*(*it));                                       \
```

Comment:
Visitor is defined one layer upper. We need no repatly create Visitor for every one iteration in this case